### PR TITLE
Fix the error of running run_pipeline.py of text_generation example

### PR DESCRIPTION
**Pull request by yuanwu2017**
# What does this PR do?

python run_pipeline.py --model_name_or_path mistralai/Mistral-7B-v0.3  --use_kv_cache --max_new_tokens 20 --prompt "He is working on"
Fix the following error:
06/09/2024 04:11:28 - INFO - __main__ - device: hpu, n_hpu: 0, bf16: False
06/09/2024 04:11:28 - INFO - __main__ - Model initialization took 10.879s
Traceback (most recent call last):
  File "/workspace/optimum-habana/examples/text-generation/text-generation-pipeline/run_pipeline.py", line 63, in <module>
    main()
  File "/workspace/optimum-habana/examples/text-generation/text-generation-pipeline/run_pipeline.py", line 44, in main
    pipe = GaudiTextGenerationPipeline(args, logger)
  File "/workspace/optimum-habana/examples/text-generation/text-generation-pipeline/pipeline.py", line 8, in __init__
    self.model, self.tokenizer, self.generation_config = initialize_model(args, logger)
ValueError: too many values to unpack (expected 3)

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by yuanwu2017** [comment]
@regisss Please help to review.

**Comment by vidyasiv** [comment]
Thanks @yuanwu2017, 
@regisss , @yuanwu2017 , do you think it would be a good idea to add a test for this fix [here](https://github.com/huggingface/optimum-habana/blob/main/tests/test_pipeline.py) ?

**Review by ssarkar2 — APPROVED** [review]

**Review by imangohari1 — CHANGES_REQUESTED** [review]
@yuanwu2017 @regisss 
Could we add a pytest for this text-gen pipeline so changes like this would actually be captured next time?

Below is a simple one that would have capture this:
```diff
From 063d8013a4f7e718e53006afd6c64d910c1d9ce3 Mon Sep 17 00:00:00 2001
From: Iman Gohari <s.m.iman.gohari@intel.com>
Date: Thu, 13 Jun 2024 22:53:31 +0000
Subject: [PATCH] fea(): Added a text-gen-piple pytest

---
 tests/test_text_generation_example.py | 49 +++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/tests/test_text_generation_example.py b/tests/test_text_generation_example.py
index dbefec2..8be2be7 100644
--- a/tests/test_text_generation_example.py
+++ b/tests/test_text_generation_example.py
@@ -2,8 +2,10 @@ import json
 import os
 import re
 import subprocess
+import sys
 from pathlib import Path
 from tempfile import TemporaryDirectory
+from unittest import TestCase

 import pytest

@@ -283,3 +285,50 @@ def test_text_generation_torch_compile(model_name: str, baseline: float, token:
 def test_text_generation_torch_compile_distributed(model_name: str, baseline: float, token: str):
     world_size = 8
     _test_text_generation(model_name, baseline, token, deepspeed=True, world_size=world_size, torch_compile=True)
+
+
+class TextGenPipeline(TestCase):
+    def test_text_generation_pipeline_script(self):
+        path_to_script = (
+            Path(os.path.dirname(__file__)).parent
+            / "examples"
+            / "text-generation"
+            / "text-generation-pipeline"
+            / "run_pipeline.py"
+        )
+
+        cmd_line = f"""ls {path_to_script}""".split()
+
+        # check find existence
+        p = subprocess.Popen(cmd_line)
+        return_code = p.wait()
+
+        # Ensure the run finished without any issue
+        self.assertEqual(return_code, 0)
+
+    def test_text_generation_pipeline_falcon(self):
+        path_to_script = (
+            Path(os.path.dirname(__file__)).parent
+            / "examples"
+            / "text-generation"
+            / "text-generation-pipeline"
+            / "run_pipeline.py"
+        )
+        # os.environ["PYTHONPATH"] += os.pathsep +
+        sys.path.append((Path(os.path.dirname(__file__)).parent / "examples" / "text-generation"))
+        cmd_line = f"""
+                 python3
+                 {path_to_script}
+                 --model_name_or_path tiiuae/falcon-7b
+                 --max_new_tokens 100
+                 --bf16
+                 --use_hpu_graphs
+                 --use_kv_cache
+                 --do_sample
+                 """.split()
+        # Run train_text_to_image_sdxl.y
+        p = subprocess.Popen(cmd_line)
+        return_code = p.wait()
+
+        # Ensure the run finished without any issue
+        self.assertEqual(return_code, 0)
--
2.34.1

```

executed as `python -m pytest  tests/test_text_generation_example.py  -v -s -k "test_text_generation_pipeline"`
```bash 
========================================================================================================= test session starts =========================================================================================================
platform linux -- Python 3.10.12, pytest-7.4.4, pluggy-1.4.0 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /root/optimum-habana
configfile: setup.cfg
collected 21 items / 19 deselected / 2 selected

tests/test_text_generation_example.py::TextGenPipeline::test_text_generation_pipeline_falcon /usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:485: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
/usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:342: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
[WARNING|utils.py:212] 2024-06-13 22:56:37,602 >> optimum-habana v1.12.0.dev0 has been validated for SynapseAI v1.16.0 but habana-frameworks v1.15.0.479 was found, this could lead to undefined behavior!
[WARNING|utils.py:225] 2024-06-13 22:56:38,349 >> optimum-habana v1.12.0.dev0 has been validated for SynapseAI v1.16.0 but the driver version is v1.15.0, this could lead to undefined behavior!
06/13/2024 22:56:38 - INFO - __main__ - Initializing text-generation pipeline...
/usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:342: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
  _torch_pytree._register_pytree_node(
/usr/local/lib/python3.10/dist-packages/transformers/deepspeed.py:23: FutureWarning: transformers.deepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
Fetching 2 files: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 417.55it/s]
Fetching 2 files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 57456.22it/s]
06/13/2024 22:56:39 - INFO - __main__ - Single-device run.
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.78it/s]
============================= HABANA PT BRIDGE CONFIGURATION ===========================
 PT_HPU_LAZY_MODE = 1
 PT_RECIPE_CACHE_PATH =
 PT_CACHE_FOLDER_DELETE = 0
 PT_HPU_RECIPE_CACHE_CONFIG =
 PT_HPU_MAX_COMPOUND_OP_SIZE = 9223372036854775807
 PT_HPU_LAZY_ACC_PAR_MODE = 1
 PT_HPU_ENABLE_REFINE_DYNAMIC_SHAPES = 0
---------------------------: System Configuration :---------------------------
Num CPU Cores : 160
CPU RAM       : 1056375104 KB
------------------------------------------------------------------------------
06/13/2024 22:56:47 - INFO - __main__ - Args: Namespace(device='hpu', model_name_or_path='tiiuae/falcon-7b', bf16=True, max_new_tokens=100, max_input_tokens=0, batch_size=1, warmup=3, n_iterations=5, local_rank=0, use_kv_cache=True, use_hpu_graphs=True, dataset_name=None, column_name=None, do_sample=True, num_beams=1, trim_logits=False, seed=27, profiling_warmup_steps=0, profiling_steps=0, profiling_record_shapes=False, prompt=None, bad_words=None, force_words=None, assistant_model=None, peft_model=None, num_return_sequences=1, token=None, model_revision='main', attn_softmax_bf16=False, output_dir=None, bucket_size=-1, bucket_internal=False, dataset_max_samples=-1, limit_hpu_graphs=False, reuse_cache=False, verbose_workers=False, simulate_dyn_prompt=None, reduce_recompile=False, use_flash_attention=False, flash_attention_recompute=False, flash_attention_causal_mask=False, flash_attention_fast_softmax=False, book_source=False, torch_compile=False, ignore_eos=True, temperature=1.0, top_p=1.0, const_serialization_path=None, disk_offload=False, trust_remote_code=False, quant_config='', world_size=0, global_rank=0)
06/13/2024 22:56:47 - INFO - __main__ - device: hpu, n_hpu: 0, bf16: True
06/13/2024 22:56:47 - INFO - __main__ - Model initialization took 8.684s
06/13/2024 22:56:47 - INFO - __main__ - Graph compilation...
06/13/2024 22:56:52 - INFO - __main__ - Running inference iteration 1...
Prompt[1][1]: DeepSpeed is a machine learning framework
Generated Text[1][1]: 'DeepSpeed is a machine learning framework that was introduced by DeepMind, the AI laboratory of Google and Alphabet. This framework was used to play a game called Go against 3 of the top Go players in the world and won with a score of 5-0. The only game that DeepMind lost was to AlphaGo Zero, the Go-playing AI that was learning how to play the game by self-teaching itself, and to Google’s own version of AlphaGo.\nThe DeepMind team had published an in'

06/13/2024 22:56:53 - INFO - __main__ - Running inference iteration 2...
Prompt[2][1]: DeepSpeed is a machine learning framework
Generated Text[2][1]: 'DeepSpeed is a machine learning framework developed at UC Berkeley by Andrew Ng and his team. At its core, DeepSpeed is a compiler for neural network functions that supports a wide range of training approaches, from GPU to TPU to NVDIA DGX. I love how the developers created a flexible API that can support multiple training architectures. One of the best features of DeepSpeed is its automatic partitioner that automatically chooses the data partitioning in different training settings. In the following, I will show you how to set up a Deep'

06/13/2024 22:56:54 - INFO - __main__ - Running inference iteration 3...
Prompt[3][1]: DeepSpeed is a machine learning framework
Generated Text[3][1]: 'DeepSpeed is a machine learning framework that is designed for efficient and scalable distributed training of deep neural networks.\nOpenCL is an open standard API for heterogeneous system architecture. It allows efficient programming of compute coprocessor that can be used to offload computation from the main processing unit.\nDeepSpeed is designed to leverage OpenCL-based accelerators for machine learning.\nCurrently, support for the following accelerators are built into DeepSpeed:\nDeepSpeed is a framework for machine learning. The main contribution of DeepSpeed is efficient'

06/13/2024 22:56:55 - INFO - __main__ - Running inference iteration 4...
Prompt[4][1]: DeepSpeed is a machine learning framework
Generated Text[4][1]: 'DeepSpeed is a machine learning framework that trains deep neural networks through an off-policy learning method that avoids the need to carefully evaluate the policies it produces..'

06/13/2024 22:56:55 - INFO - __main__ - Running inference iteration 5...
Prompt[5][1]: DeepSpeed is a machine learning framework
Generated Text[5][1]: 'DeepSpeed is a machine learning framework that is aimed at training on large scale datasets like those that arrive with deep learning approaches for computer vision. Since its public launch, it has been updated almost every week.\nIn a blog post on Wednesday, the DeepSpeed team shared a new set of improvements and optimizations and announced that the project is now open-sourcing.\n“DeepSpeed is a Python library that builds on the existing research and experience in deep learning to help you train neural networks on data sets that are too large and'

Inference Duration (for 5 iterations): 3.8097385700057202 seconds
Throughput: 131.24260124737359 tokens/second
PASSED
tests/test_text_generation_example.py::TextGenPipeline::test_text_generation_pipeline_script /root/optimum-habana/examples/text-generation/text-generation-pipeline/run_pipeline.py
PASSED

========================================================================================================== warnings summary ===========================================================================================================
../../usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:485
  /usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:485: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
    _torch_pytree._register_pytree_node(

../../usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:342
  /usr/local/lib/python3.10/dist-packages/transformers/utils/generic.py:342: UserWarning: torch.utils._pytree._register_pytree_node is deprecated. Please use torch.utils._pytree.register_pytree_node instead.
    _torch_pytree._register_pytree_node(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================ 2 passed, 19 deselected, 2 warnings in 25.18s ============================================================================================
```

**Comment by yuanwu2017** [comment]
@imangohari1 @vidyasiv  
OK. I will add the testcase.

**Comment by imangohari1** [comment]
> @imangohari1 @vidyasiv OK. I will add the testcase.

Thank you. 
I suggest we wait for @regisss  recommendation on what test is needed here, but you can apply the one I provided with the file attached and then `git am < 0001*.patch `
[0001-fea-Added-a-text-gen-piple-pytest.patch](https://github.com/user-attachments/files/15830952/0001-fea-Added-a-text-gen-piple-pytest.patch)

**Comment by yuanwu2017** [comment]
@imangohari1 @regisss   Done.

File: tests/test_text_generation_example.py [hunk 1]
```diff
@@ -283,3 +285,50 @@ def test_text_generation_torch_compile(model_name: str, baseline: float, token:
...
+
+    def test_text_generation_pipeline_falcon(self):
+        path_to_script = (
+            Path(os.path.dirname(__file__)).parent
+            / "examples"
+            / "text-generation"
+            / "text-generation-pipeline"
+            / "run_pipeline.py"
+        )
+        # os.environ["PYTHONPATH"] += os.pathsep +
```
Comment (regisss) [inline tests/test_text_generation_example.py:317]: Commented line to remove

File: tests/test_text_generation_example.py [hunk 2]
```diff
@@ -283,3 +285,50 @@ def test_text_generation_torch_compile(model_name: str, baseline: float, token:
...
+                 python3
+                 {path_to_script}
+                 --model_name_or_path tiiuae/falcon-7b
+                 --max_new_tokens 100
+                 --bf16
+                 --use_hpu_graphs
+                 --use_kv_cache
+                 --do_sample
+                 """.split()
+        # Run train_text_to_image_sdxl.y
```
Comment (regisss) [inline tests/test_text_generation_example.py:329]: Wrong comment, let's remove it

**Review by regisss — COMMENTED** [review]
LGTM!
I just left a couple of comments to remove some commented lines and then I'll merge.

**Review by regisss — APPROVED** [review]

**Pull request merged**